File: conversions/reverse-string/elide-reverse-string.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-reverse-string
#
# This demonstrates calling the TypeScript reverse-string implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One reverse-string implementation shared across TypeScript and Ruby
# - Consistent reverse a string across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript reverse-string ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# reverse_string_module = Elide.require('./elide-reverse-string.ts')

# Example: Basic usage
# result = reverse_string_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses reverse a string"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for text manipulation, palindrome checking, data reversal"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide reverse-string (TypeScript)         │"
puts "│   conversions/reverse-string/elide-reverse-string.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same reverse a string everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/reverse_string.rb"
puts "  REVERSE_STRING = Elide.require('./elide-reverse-string.ts')"

File: converted/utilities/reverse-string/elide-reverse-string.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-reverse-string

This demonstrates calling the TypeScript reverse-string implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One reverse-string implementation shared across TypeScript and Python
- Consistent reverse a string across services
- No Python custom implementation needed
- Perfect for text manipulation, palindrome checking, data reversal
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# reverse_string_module = require('./elide-reverse-string.ts')

print("=== Python Calling TypeScript reverse-string ===\n")

# Example: Basic usage
# result = reverse_string_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses reverse a string")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for text manipulation, palindrome checking, data reversal")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide reverse-string (TypeScript)         │")
print("│   conversions/reverse-string/elide-reverse-string.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same reverse a string everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  reverse_string = require('./elide-reverse-string.ts')")
print("  # Use it directly!")
